server/spec/requests/api/v1/mqtt_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Api::V1::MqttController', type: :request do
+  describe 'POST #publish' do
+    it 'publishes light on to the MQTT broker' do
+      client = double(MQTT::Client)
```

Comment:
Can you walk me through what is `double(MQTT::Client)` doing here? Is it just to instantiate a class instance, eg an MQTT client? Why not just instantiate it directly?

Reply:
Yesssir for sure

The idea of mocking is so that we don't have to actually call the external library! If our broker is broken then our app is down and it's kind of out of our control. All we can do is angrily call them and be like "hey I need your service to come back up because we need it"

SO, there's not much of a point for the test to test the external library. Instead, we stub out the call to the external library, and make a fake version of that which and perform tests upon that thing instead. Later on in the test we basically just test that `client` is called with the right parameters, which tests the functionality of our app